rest_framework/pagination.py

```diff
@@ -201,7 +201,34 @@ def to_html(self):  # pragma: no cover
         raise NotImplementedError('to_html() must be implemented to display page controls.')
 
 
-class PageNumberPagination(BasePagination):
+class PageSizePaginationMixin(object):
```

Comment:
> I would avoid mixins wherever possible

Do you have any Ideas how to do that without mixins? `CursorPagination` works absolutely fine with user defined limits, even when limits changed for sequential requests.